packages/gatsby-source-contentful/src/gatsby-node.js

```diff
@@ -262,11 +262,4 @@ exports.onPreExtractQueries = async ({ store, getNodesByType }) => {
   if (gatsbyImageDoesNotExist) {
     return
   }
-
-  // We have both gatsby-image installed as well as ImageSharp nodes so let's
-  // add our fragments to .cache/fragments.
-  await fs.copy(
-    require.resolve(`gatsby-source-contentful/src/fragments.js`),
-    `${program.directory}/.cache/fragments/contentful-asset-fragments.js`
-  )
```

Comment:
Can we remove more from this file? Seems like after removing this part only thing we need is to ensure `CACHE_DIR` and we could remove code from `if (getNodesByType(`ContentfulAsset`).length == 0) {` onwards?

Reply:
Excellent point! 

We should be able to remove everything except ensuring `cache/contentful/assets` exists.

Reply:
good catch @pieh  - updated the PR.